infraestructure/services/services.go

```diff
@@ -0,0 +1,85 @@
+package services
+
+//This file manages the dependency injection system
+
+import (
+	"fmt"
+	"log"
+	"reflect"
+)
+
+//ServiceRegistry is the structure of the ServicesRegistry
+type ServiceRegistry struct {
+	services     map[reflect.Type]Service
+	serviceTypes []reflect.Type
+}
+
+//ServicesRegistry is the global variable for store and fetch services
+var ServicesRegistry *ServiceRegistry
+
+//NewServiceRegistry returns the instance of the ServiceRegistry structure.
+func NewServiceRegistry() *ServiceRegistry {
+	ServicesRegistry = &ServiceRegistry{
+		services: make(map[reflect.Type]Service),
+	}
+	return ServicesRegistry
+}
+
+const (
+	STATUSOK     = iota
+	STATUSSTOPED = iota
+)
```

Comment:
```suggestion
type Status int

const (
        A Status = iota
        C
        T
        G
)
```

We can be a bit more idiomatic here, you just need to refine the type of Status.